services/expiration/expiration-notification.server.ts

```diff
@@ -54,24 +45,45 @@ export function init() {
   });
 
   // worker instance to process DNS Record/Certificates expiration notification jobs
-  expirationNotificationWorker = new Worker<ExpirationStatusPayload>(
+  expirationNotificationWorker = new Worker(
     expirationNotificationQueueName,
     async (job) => {
       const { type } = job.data;
       try {
         logger.debug(`Notifications: processing job ${job.name}`);
-        let records = await getRecordsByExpiration(RecordType.DnsRecord);
+        let dnsRecords = await getExpiringDnsRecords();
+        await Promise.all(
+          dnsRecords.map(async ({ id, subdomain, expiresAt, lastNotified, user }) => {
+            if (!lastNotified || lastNotified < dayjs().subtract(30, 'd').toDate()) {
+              await updateStatusAndNotify(type, id, {
+                emailAddress: user.email,
+                subject: 'My.Custom.Domain DNS record approaching expiration',
+                message: `${
+                  user.displayName
+                }, this is a friendly reminder that your DNS record with subdomain: ${subdomain} will expire on: ${expiresAt.toLocaleDateString(
+                  'en-CA'
+                )}. Log in to My.Custom.Domain to renew.`,
+              });
+            }
+          })
+        );
+        let certificates = await getExpiringCertificates();
         await Promise.all(
-          records.map(async (record) => {
-            if (!record.lastNotified || record.lastNotified < dayjs().subtract(30, 'd').toDate()) {
-              await updateStatusAndNotify(type, record.id, {
-                emailAddress: record.user.email,
-                subject: `Sample ${type} expiration notice`,
-                message: `Sample ${type} expiration notice message`,
+          certificates.map(async ({ id, domain, validTo, lastNotified, user }) => {
+            if (!lastNotified || lastNotified < dayjs().subtract(30, 'd').toDate()) {
```

Comment:
Is this needed here when we also check a similar or condition in getExpiringCertificates?

Reply:
I was hesitant to remove it because it was added in a previous PR. Great catch.